test/ProjectTemplates/Microsoft.Extensions.AI.Templates.IntegrationTests/AIChatWebExecutionTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Extensions.AI.Templates.Tests;
+
+public class AIChatWebExecutionTests : TemplateExecutionTestBase<AIChatWebExecutionTests>, ITemplateExecutionTestConfigurationProvider
+{
+    public AIChatWebExecutionTests(TemplateExecutionTestFixture fixture, ITestOutputHelper outputHelper)
+        : base(fixture, outputHelper)
+    {
+    }
+
+    public static TemplateExecutionTestConfiguration Configuration { get; } = new()
+    {
+        TemplatePackageName = "Microsoft.Extensions.AI.Templates",
+        TestOutputFolderPrefix = "AIChatWeb"
+    };
+
+    [Theory]
+    [InlineData("AzureApp",
+        "--provider", "azureopenai",
+        "--vector-store", "azureaisearch")]
+    [InlineData("AzureAppManagedIdentity",
+        "--provider", "azureopenai",
+        "--vector-store", "azureaisearch",
+        "--managed-identity")]
+    [InlineData("GitHubModelsWithLocalStore",
+        "--provider", "githubmodels",
+        "--vector-store", "local")]
+    [InlineData("OllamaWithLocalStore",
+        "--provider", "ollama",
+        "--vector-store", "local")]
+    [InlineData("OpenAIWithLocalStore",
+        "--provider", "openai",
+        "--vector-store", "local")]
+    public async Task CreateRestoreAndBuild_BasicTemplate(string projectName, params string[] args)
+    {
+        var project = await Fixture.CreateProjectAsync(
+            templateName: "aichatweb",
+            projectName,
+            args);
+
+        await Fixture.RestoreProjectAsync(project);
+        await Fixture.BuildProjectAsync(project);
+    }
+
+    [Theory]
+    [InlineData("AzureApp",
+        "--provider", "azureopenai",
+        "--vector-store", "azureaisearch")]
+    [InlineData("GitHubModelsWithQdrant",
+        "--provider", "githubmodels",
+        "--vector-store", "qdrant")]
+    [InlineData("OllamaWithLocalStore",
+        "--provider", "ollama",
+        "--vector-store", "local")]
+    [InlineData("OpenAIWithLocalStore",
+        "--provider", "ollama",
+        "--vector-store", "local")]
```

Comment:
I think there are other configuration combinations possible/supported that aren't covered; is that right? Is this set of scenarios giving us the distinct set of possible dependencies?

I'd be inclined to generate every possible/supported configuration and doing so here in a way where when new combinations are added, it's less likely we'll forget to add the relevant coverage. That could be achieved by using a MemberData approach that generates all possible combinations of providers, vector stores, and the other switches, when the test having logic to identify unsupported combinations and skip the scenario.

Reply:
Generating all valid combinations sounds good 🙂 I initially avoided that to prevent redundant checks from taking up too much time on CI. But since each check is only ~10 seconds, I think this should be fine. I'll go ahead and implement that approach.